dora/core/common/src/main/java/alluxio/PositionReader.java

```diff
@@ -103,4 +104,26 @@ int readInternal(long position, ReadTargetBuffer buffer, int length)
    * Closes the positon reader and do cleanup job if any.
    */
   default void close() throws IOException {}
+
+  /**
+   * Extract from read internal of different ufs position reader.
+   * read data from ufs
+   * @param in input stream
+   * @param buffer target byte buffer
+   * @param bytesToRead bytes to read
+   * @return bytes read, or -1 none of data is read
+   */
+  default int readDataInternal(InputStream in, ReadTargetBuffer buffer,
```

Comment:
why is this a default impl in `PositionReader`? The method reads from a `InputStream`, has nothing to do with `PositionReader`.

Reply:
I think what you need is something that is the equivalent of `MultiRangeObjectInputStream` but for `PositionReader`.